src/interval.c

```diff
@@ -109,8 +109,33 @@ ts_interval_from_sql_input(Oid relid, Datum interval, Oid interval_type, const c
 		elog(ERROR, "internal error: no open dimension found while parsing interval");
 
 	partitioning_type = ts_dimension_get_partition_type(open_dim);
+	if (IS_INTEGER_TYPE(partitioning_type))
+	{
+		if (strlen(NameStr(open_dim->fd.integer_now_func)) == 0 ||
+			strlen(NameStr(open_dim->fd.integer_now_func_schema)) == 0)
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
```

Comment:
I suggest a different error code, since this one is reserved for invalid parameter values when calling (user-facing) SQL functions. For instance, user calls `SELECT set_port(port => 134000);` then the value for `port` is invalid (outside the port range). 

 Also, in my understanding, having a NULL integer now func is valid.

I suggest INTERNAL_ERROR or just a simple `elog`.

Reply:
I believe this code path is exercised by drop_chunks_policy. So this is a user facing error.

Reply:
My point is that `integer_now_func` and `integer_now_func_schema` aren't, AFAIK, set as part of those functions, like in my `set_port` example. Further, the error is not about a function parameter but about some values of internal metadata being NULL.

So, I am not sure it makes sense to raise `ERRCODE_INVALID_PARAMETER_VALUE`. The error should tell you that the system is in the wrong state for the given operation.

Reply:
These are errors for the "older_than" parameter for the policy. There is a specific action needed on the user's part to fix this issue. older_than parameter on integer-time based  hypertables doesn't make sense without specifying a now() function. INTERNAL_ERROR to me typically signifies something unexpected happening in the code which is probably a bug.